controllers/reaper/reaper_controller.go

```diff
@@ -248,6 +298,29 @@ func (r *ReaperReconciler) reconcileDeployment(
 	return ctrl.Result{}, nil
 }
 
+func getDeploymentReplicas(actualDeployment client.Object, logger logr.Logger) int32 {
+	switch actual := actualDeployment.(type) {
+	case *appsv1.Deployment:
+		return *actual.Spec.Replicas
+	case *appsv1.StatefulSet:
+		return *actual.Spec.Replicas
+	default:
+		logger.Error(fmt.Errorf("unexpected type %T", actualDeployment), "Failed to get deployment replicas")
+		return math.MaxInt32
+	}
+}
+
+func forceSingleReplica(desiredDeployment *client.Object, logger logr.Logger) {
+	switch desired := (*desiredDeployment).(type) {
+	case *appsv1.Deployment:
+		desired.Spec.Replicas = ptr.To[int32](1)
+	case *appsv1.StatefulSet:
+		desired.Spec.Replicas = ptr.To[int32](1)
+	default:
+		logger.Error(fmt.Errorf("unexpected type %T", desiredDeployment), "Failed to set deployment replicas")
```

Comment:
**Suggestion:** same here, an error should be returned.

Reply:
Pushed a commit solving this.